### PR TITLE
HKG: Car Port for Hyundai Palisade and Kia Telluride 2023-24 (non-HDA2)

### PR DESCRIPTION
**Pull request by sunnyhaibin**
**Checklist**
- [x] added entry to CarInfo in selfdrive/car/*/values.py and ran `selfdrive/car/docs.py` to generate new docs
- [x] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)
- [x] route with stock system:
  - Hyundai Palisade 2023: `f2cf0de0119d2a16|2023-03-02--09-39-50`
  - Kia Telluride 2023:
- [x] route with openpilot:
  - Hyundai Palisade 2023 (non-HDA2): `f2cf0de0119d2a16|2023-03-03--21-13-05`
  - Kia Telluride 2023 (non-HDA2):
- [x] harness type:
  - Hyundai Palisade 2023:
    - [Hyundai A](https://github.com/commaai/openpilot/wiki/Hyundai-Kia-Genesis#harness-guide) (non-HDA2) ✅
  - Kia Telluride 2023:
    - [Hyundai L](https://github.com/commaai/openpilot/wiki/Hyundai-Kia-Genesis#harness-guide) (non-HDA2) ✅
- [x] architecture: CAN-FD

**Link to all physical parts during installation**
  - [comma 3X](https://comma.ai/shop/comma-3x)
  - [car harness kit](https://comma.ai/shop/car-harness)
    - Hyundai Palisade 2023:
      - [Hyundai A](https://github.com/commaai/openpilot/wiki/Hyundai-Kia-Genesis#harness-guide) (non-HDA2)
    - Kia Telluride 2023:
      - [Hyundai L](https://github.com/commaai/openpilot/wiki/Hyundai-Kia-Genesis#harness-guide) (non-HDA2)

**Software URL**
- https://bit.ly/palisade-2023-no-hda2

**Requires**
- https://github.com/commaai/panda/pull/1267
- https://github.com/commaai/opendbc/pull/784
- https://github.com/commaai/openpilot/pull/29695
- https://github.com/commaai/openpilot/pull/29733

**Resolves**
- https://github.com/commaai/openpilot/issues/26571

**Related**
- https://github.com/commaai/openpilot/pull/27392
- https://github.com/commaai/panda/pull/1243

Thanks to community Hyundai Palisade 2023 owner `SKILLED#7567` (Discord).

### CODE REVIEW

**Comment by sunnyhaibin** [comment]
Steering works in the latest commits 🎉
`f2cf0de0119d2a16|2023-03-03--20-22-30`

image

**Comment by Guardiola31337** [comment]
Hey @sunnyhaibin we should probably update the description to reflect the correct harness type - as we did in the other PR https://github.com/commaai/openpilot/pull/27392#issuecomment-1444802253 👍

**Comment by sunnyhaibin** [comment]
> Hey @sunnyhaibin we should probably update the description to reflect the correct harness type - as we did in the other PR [#27392 (comment)](https://github.com/commaai/openpilot/pull/27392#issuecomment-1444802253) 👍

They should already be the correct harness type. HDA2 and non-HDA2 variants use different harness configurations.

**Comment by Guardiola31337** [comment]
@sunnyhaibin Noob question: What's the difference between HDA and non-HDA? I'm asking because I want to get the correct harness type for my Kia Telluride 2023 😅

**Comment by Guardiola31337** [comment]
After researching a bit it seems that depends on the trim. HDA2 is for SX and up. My model is S so I probably need the non-HDA2, could you confirm?

**Comment by sunnyhaibin** [comment]
> After researching a bit it seems that depends on the trim. HDA2 is for SX and up. My model is S so I probably need the non-HDA2, could you confirm?

If your trim is S, then your car has HDA 1.5 according to Kia's website, so it would be a non-HDA2 configuration. https://www.kia.com/us/en/telluride/specs

**Comment by sunnyhaibin** [comment]
> I would like to test this I have a 2023SX Kia telluride have the, three and the harness. What do I need to do to install? I don't see the installer link for this?

Is this car HDA1/1.5 or HDA2? Required hardware is listed in the description of this PR.

**Comment by Scpcguy** [comment]
> > I would like to test this I have a 2023SX Kia telluride have the, three and the harness. What do I need to do to install? I don't see the installer link for this?
> 
> Is this car HDA1/1.5 or HDA2? Required hardware is listed in the description of this PR.

I have the hardware - just not sure about getting the software.  I have installed the standard Sunny - "fork" but there is not the option for 2023 sx hda2.

**Comment by sunnyhaibin** [comment]
> > > I would like to test this I have a 2023SX Kia telluride have the, three and the harness. What do I need to do to install? I don't see the installer link for this?
> > 
> > 
> > Is this car HDA1/1.5 or HDA2? Required hardware is listed in the description of this PR.
> 
> I have the hardware - just not sure about getting the software. I have installed the standard Sunny - "fork" but there is not the option for 2023 sx hda2.

HDA2:
https://bit.ly/sp-palisade-2023-port

**Comment by Scpcguy** [comment]
I have been testing - I don't know if I can offer data to you or not but would be glad to do so.  The braking is a bit hard and quick.  Keeping the car in the lanes and driving without interaction is working fine as regards to just straight driving. I do not see the TRIANGLE icon on the Comma and evidently is not interfacing with the warning of other cars etc if you want to change lanes.  Please let me know what I can do to get you data if needed. Kia Telluride 2023 SX - HDA2.

File: selfdrive/car/hyundai/values.py [hunk 1]
```diff
@@ -62,6 +67,7 @@ class HyundaiFlags(IntFlag):
   CANFD_ALT_GEARS_2 = 64
   SEND_LFA = 128
   USE_FCA = 256
+  CAN_CANFD = 512
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:70]: Let's use the existing `CANFD_HDA2` for deciding HDA2-specific behavior, but just add a new flag for the half-CANFD cars

File: selfdrive/car/hyundai/values.py [hunk 2]
```diff
@@ -88,6 +94,7 @@ class CAR:
   SONATA_LF = "HYUNDAI SONATA 2019"
   TUCSON = "HYUNDAI TUCSON 2019"
   PALISADE = "HYUNDAI PALISADE 2020"
+  PALISADE_2023 = "HYUNDAI PALISADE 2023"
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:97]: new generation or no?

File: selfdrive/car/hyundai/values.py (hunk 1 above)
Comment (sshane) [inline selfdrive/car/hyundai/values.py:70]: Let's use the existing `CANFD_HDA2` for deciding HDA2-specific behavior, but just add a new flag for the half-CANFD cars (so you don't need two flags for the Palisades). Think it makes sense, as these cars are using a CAN FD bus, doesn't matter which messages they're using

File: selfdrive/car/hyundai/interface.py [hunk 3]
```diff
@@ -45,8 +45,13 @@ def _get_params(ret, candidate, fingerprint, car_fw, experimental_long):
         if candidate not in CANFD_RADAR_SCC_CAR:
           ret.flags |= HyundaiFlags.CANFD_CAMERA_SCC.value
     else:
+      # detect non-HDA2 with CAN and CAN-FD definitions
```
Comment (sshane) [inline selfdrive/car/hyundai/interface.py:48]: I think it makes sense to add them to CANFD list, and dynamically detect if they are using CAN messages by message presence

**Review by sshane — CHANGES_REQUESTED** [review]

File: selfdrive/car/hyundai/values.py (hunk 2 above)
Comment (sshane) [reply selfdrive/car/hyundai/values.py:97]: oh platform code is the same :( this means initial idea for fuzzy FP won't work with these cars just yet

Comment (sshane) [reply selfdrive/car/hyundai/values.py:97]: in the future, we should be able to combine the two Palisade platforms and detect all the required features (if CAN FD bus, if CAN FD bus and CAN messages, etc), but that will require a bigger chance since we rely on the CAN FD list for static docs generation and likely other places too.

File: selfdrive/car/hyundai/values.py [hunk 4]
```diff
@@ -181,8 +188,12 @@ def init_make(self, CP: car.CarParams):
     HyundaiCarInfo("Hyundai Palisade 2020-22", "All", video_link="https://youtu.be/TAnDqjF4fDY?t=456", car_parts=CarParts.common([CarHarness.hyundai_h])),
     HyundaiCarInfo("Kia Telluride 2020-22", "All", car_parts=CarParts.common([CarHarness.hyundai_h])),
   ],
+  CAR.PALISADE_2023: [
+    HyundaiCarInfo("Hyundai Palisade (without HDA II) 2023", "All", car_parts=CarParts.common([CarHarness.hyundai_a])),
+    HyundaiCarInfo("Kia Telluride (without HDA II) 2023", "All", car_parts=CarParts.common([CarHarness.hyundai_l])),
+  ],
   CAR.VELOSTER: HyundaiCarInfo("Hyundai Veloster 2019-20", min_enable_speed=5. * CV.MPH_TO_MS, car_parts=CarParts.common([CarHarness.hyundai_e])),
-  CAR.SONATA_HYBRID: HyundaiCarInfo("Hyundai Sonata Hybrid 2020-22", "All", car_parts=CarParts.common([CarHarness.hyundai_a])),
+  CAR.SONATA_HYBRID: HyundaiCarInfo("Hyundai Sonata Hybrid 2020-22", "All", car_parts=CarParts.common([CarHarness.hyundai_e])),
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:196]: why was this changed?

File: selfdrive/car/hyundai/values.py (hunk 2 above)
Comment (sshane) [reply selfdrive/car/hyundai/values.py:97]: Apparently this is a face lifted car, wanna call it that?

> The updated Palisade debuted for the North American market at the [New York International Auto Show](https://en.wikipedia.org/wiki/New_York_International_Auto_Show#2022) on April 13, 2022, for the 2023 model year.[[23]](https://en.wikipedia.org/wiki/Hyundai_Palisade#cite_note-23)[[24]](https://en.wikipedia.org/wiki/Hyundai_Palisade#cite_note-24) It features a revised front end, and an enhanced infotainment system with 12-inch navigation.[[25]](https://en.wikipedia.org/wiki/Hyundai_Palisade#cite_note-25)

https://en.wikipedia.org/wiki/Hyundai_Palisade#Facelift

File: selfdrive/car/hyundai/values.py (hunk 4 above)
Comment (sunnyhaibin) [reply selfdrive/car/hyundai/values.py:196]: I might have accidentally changed it when the generator ran. Will fix it during rebase.

File: selfdrive/car/hyundai/values.py (hunk 2 above)
Comment (sunnyhaibin) [reply selfdrive/car/hyundai/values.py:99]: > Apparently this is a face lifted car, wanna call it that?
> 
> > The updated Palisade debuted for the North American market at the [New York International Auto Show](https://en.wikipedia.org/wiki/New_York_International_Auto_Show#2022) on April 13, 2022, for the 2023 model year.[[23]](https://en.wikipedia.org/wiki/Hyundai_Palisade#cite_note-23)[[24]](https://en.wikipedia.org/wiki/Hyundai_Palisade#cite_note-24) It features a revised front end, and an enhanced infotainment system with 12-inch navigation.[[25]](https://en.wikipedia.org/wiki/Hyundai_Palisade#cite_note-25)
> 
> https://en.wikipedia.org/wiki/Hyundai_Palisade#Facelift

@sshane Like this?
`PALISADE_FACE_LIFT = "HYUNDAI PALISADE FACE LIFT"`

**Comment by sshane** [comment]
We've moved the car interfacing code to our [`opendbc`](https://github.com/commaai/opendbc) repository, which is now the new home for car ports and fingerprints. Please re-open your pull request against [`opendbc`](https://github.com/commaai/opendbc) at your convenience by using this command below. This will transform all changes under `selfdrive/car/` to `opendbc_repo/opendbc/car/`. Make sure you have initialized submodules beforehand and have checked out the latest `opendbc` commit.

```bash
PR_NUMBER=33045
curl -L https://github.com/commaai/openpilot/pull/$PR_NUMBER.patch | sed -e 's/selfdrive\/car/opendbc_repo\/opendbc\/car/g' | git apply -v --reject
```

Simply replace the PR number with your own. Once done, add the files, fix any conflicts, and open a new PR. Alternatively, you may start a new PR from scratch if that is easier for you.
